CMakeLists.txt

```diff
@@ -65,12 +65,8 @@ include(GenerateExportHeader)
 find_package(Boost 1.67 REQUIRED)
 find_package(TBB CONFIG)
 if(TBB_FOUND)
-  set(ENABLE_THREAD_LIMIT
-      OFF
-      CACHE
-        BOOL
-        "Limit the number TBB threads. Exact number to be specified via THREAD_LIMIT"
-  )
+  option(ENABLE_THREAD_LIMIT
+          "Limit the number TBB threads. Limit number set via THREAD_LIMIT" OFF)
```

Comment:
Question for my education: In which cases should `option` be used vs. `set`?

Reply:
If it's not a bool use `set`. I think option also has some additional features for chaining (enabling options).